### ISSUE TITLE
[C++][CMake][Windows] Can not compile, the project failed with an unknow reason (vs17)

### ISSUE THREAD

**Issue by SuperFola** [antlr/antlr4#2225]
Hi,

I am using ANTLR4 (or at least I am trying to) for a small (shitty) personnal project and even though I checked my CMakeLists.txt more times than enough (as well as some friend of mine), the compilation is failing with an unknown reason under visual studio 2017 (Windows 10).

I followed up all the steps given in http://blorente.me//Antlr-,-C++-and-CMake-Wait-what.html and double check my CMakeLists.txt/.cmake files using the repo given, but still can not find why it does not want to compile.

(for those who are interested in trying themselves to compile my project : https://github.com/SuperFola/Kafe, but I have clearly the same files as the ones on the given github repo, so it shouldn't be a problem).

It sounds a bit like a question about "how to use ANTLR", but if anyone could figure out why it's not compiling we could update the blog post which is not very handy :/

- [ X ] I am not submitting a question on how to use ANTLR; instead, go to [antlr4-discussion google group](https://groups.google.com/forum/#!forum/antlr-discussion) or ask at [stackoverflow](http://stackoverflow.com/questions/tagged/antlr4)
- [ X ] I have done a search of the existing issues to make sure I'm not sending in a duplicate

**Comment by EternalPhane** [comment]
I've tried to compile your project, and got only **one** error:
![screenshot_1](https://user-images.githubusercontent.com/11267872/36340435-ce7d68f4-1417-11e8-9381-c0f274659f71.png)

Digging a little deeper, I found your linker additional dependencies is wierd:
![screenshot_2](https://user-images.githubusercontent.com/11267872/36340436-d5998280-1417-11e8-8fd7-d921ca8a51fb.png)
There're two `antlr4-runtime.lib`s, who cause the error `LNK1104: cannot open file 'antlr4-runtime.lib'`.

Obviously, you've got duplicate `target_link_libraries`s in your CMakeLists.txt: 
https://github.com/SuperFola/Kafe/blob/3cff29379e39dd3ec77caf884519fe413df2af85/src/CMakeLists.txt#L14-L17
So I built again with line 17 commented. This time it compiled as expected.
Hope this could help you.
